llvm/lib/Transforms/Instrumentation/SoftPointerAuth.cpp

```diff
@@ -276,32 +278,40 @@ Value *SoftPointerAuth::emitResign(IRBuilderTy &builder, Value *pointer,
 
 Value *SoftPointerAuth::emitAuth(IRBuilderTy &builder, Value *pointer,
                                  Value *key, Value *discriminator) {
-  auto call = builder.CreateCall(getAuthPointerFn(),
-                                 { pointer, key, discriminator });
+  auto callee = getAuthPointerFn();
+  auto call = builder.CreateCall(
+      cast<FunctionType>(callee->getType()->getPointerElementType()), callee,
+      {pointer, key, discriminator});
   call->setDoesNotThrow();
   return call;
 }
 
 Value *SoftPointerAuth::emitStrip(IRBuilderTy &builder, Value *pointer,
                                   Value *key) {
-  auto call = builder.CreateCall(getStripPointerFn(),
-                                 { pointer, key });
+  auto callee = getStripPointerFn();
+  auto call = builder.CreateCall(
+      cast<FunctionType>(callee->getType()->getPointerElementType()), callee,
+      {pointer, key});
   call->setDoesNotThrow();
   return call;
 }
 
 Value *SoftPointerAuth::emitBlend(IRBuilderTy &builder, Value *primary,
                                   Value *secondary) {
-  auto call = builder.CreateCall(getBlendDiscriminatorFn(),
-                                 { primary, secondary });
+  auto callee = getBlendDiscriminatorFn();
+  auto call = builder.CreateCall(
+      cast<FunctionType>(callee->getType()->getPointerElementType()), callee,
+      {primary, secondary});
   call->setDoesNotThrow();
   return call;
 }
 
 Value *SoftPointerAuth::emitSignGeneric(IRBuilderTy &builder, Value *value,
                                         Value *discriminator) {
-  auto call = builder.CreateCall(getSignGenericFn(),
-                                 { value, discriminator });
+  auto callee = getSignGenericFn();
+  auto call = builder.CreateCall(
+      cast<FunctionType>(callee->getType()->getPointerElementType()), callee,
+      {value, discriminator});
```

Comment:
All of the `getWhateverFn()` helpers should return `FunctionCallee`, which should be straightforward.

Reply:
Thanks for the suggestion and for `llvm::PointerType` explanation, the change looks cleaner now.